### PR TITLE
Only run FV on new or updated specs

### PR DESCRIPTION
**Pull request by Amxx**
Fixes LIB-788

### CODE REVIEW

**Comment by Amxx** [comment]
So the FV fails because this branch comes from my repo, and we are missing the certora apikey. However, the good thing is that it fails only once, for Ownable (which was modified in the PR). Other specs are not run.

image

File: certora/run.js [hunk 1]
```diff
@@ -1,37 +1,65 @@
...
 const { PassThrough } = require('stream');
 const events = require('events');
-const limit = require('p-limit')(MAX_PARALLEL);
 
-let [, , request = '', ...extraOptions] = process.argv;
-if (request.startsWith('-')) {
-  extraOptions.unshift(request);
-  request = '';
+const argv = require('yargs')
+  // .env('')
```
Comment (frangio) [inline certora/run.js:15]:
```suggestion
```

File: certora/run.js [hunk 2]
```diff
@@ -1,37 +1,65 @@
...
-  const [reqSpec, reqContract] = request.split(':').reverse();
-  specs = Object.values(specs).filter(s => reqSpec === s.spec && (!reqContract || reqContract === s.contract));
-  if (specs.length === 0) {
-    console.error(`Error: Requested spec '${request}' not found in specs.json`);
-    process.exit(1);
+const specs = require(argv.spec).filter(s => argv.all || argv._.some(r => match(s, r)));
+const limit = require('p-limit')(argv.parallel);
+
+if (argv._.length == 0 && !argv.all) {
+  console.error(`Warning: No specs requested. Did you forgot to toggle '--all'`);
```
Comment (frangio) [inline certora/run.js:47]:
```suggestion
  console.error(`Warning: No specs requested. Did you forgot to toggle '--all'?`);
```

File: certora/run.js [hunk 3]
```diff
@@ -1,37 +1,65 @@
...
+    parallel: {
+      alias: 'p',
+      type: 'number',
+      default: 4,
+    },
+    options: {
+      alias: 'o',
+      type: 'array',
+      default: [],
+    },
```
Comment (frangio) [inline certora/run.js:35]: This doesn't seem to work as expected. E.g., `-o --optimistic_loop` results in `{ options: [], optimistic_loop: true }`.

We may need a special way to run a single spec with options...

Comment (Amxx) [reply certora/run.js:35]: I'll check that !

maybe `-o "--optimistic_loop"` is the way to go

Comment (Amxx) [reply certora/run.js:35]: Note that
```
node ./certora/run.js -o=--optimistic_loop -o=--optimistic_hashing ERC20
```
will not work and that you should do either
```
node ./certora/run.js ERC20 -o=--optimistic_loop -o=--optimistic_hashing
node ./certora/run.js -o=--optimistic_loop -o=--optimistic_hashing -- ERC20
```

Comment (frangio) [reply certora/run.js:35]: Ok, I don't think this is a good interface but let's just merge it and see how it works for us.

**Review by frangio — APPROVED** [review]

**Comment by frangio** [comment]
@Amxx Please update the PR title according to the guidelines before merging.

**Comment by frangio** [comment]
Verification seems to be failing with:

```
CRITICAL: Encountered an error running Certora Prover:
You are using a demo version of the tool in a CI environment.
                     If you have a premium Certora key, please set it as the environment variable CERTORAKEY.
```

**Pull request merged**